Sources/NautilusTelemetry/Utilities/FlushTimer.swift

```diff
@@ -8,25 +8,32 @@ struct FlushTimer {
 	// MARK: Lifecycle
 
 	init(flushInterval: TimeInterval, handler: @escaping () -> Void) {
+		flushTimer = DispatchSource.makeTimerSource(flags: [], queue: NautilusTelemetry.queue)
 		self.flushInterval = flushInterval
 		self.handler = handler
+		// didSet doesn't run in init
+		setupTimer()
 	}
 
 	// MARK: Internal
 
 	var handler: () -> Void
 
-	lazy var flushTimer: DispatchSourceTimer = DispatchSource.makeTimerSource(flags: [], queue: NautilusTelemetry.queue)
+	let flushTimer: DispatchSourceTimer
 
 	var flushInterval: TimeInterval {
 		didSet {
-			flushTimer.setEventHandler(handler: handler)
-			flushTimer.schedule(
-				deadline: DispatchTime.now() + flushInterval,
-				repeating: flushInterval,
-				leeway: DispatchTimeInterval.milliseconds(100)
-			)
-			flushTimer.activate()
+			setupTimer()
 		}
 	}
+
+	func setupTimer() {
+		flushTimer.setEventHandler(handler: handler)
+		flushTimer.schedule(
+			deadline: DispatchTime.now() + flushInterval,
+			repeating: flushInterval,
+			leeway: DispatchTimeInterval.milliseconds(100)
+		)
+		flushTimer.activate()
+	}
```

Comment:
@ladd is there any cleanup that we need to do of the old if we set a new flush interval? Or does this code handle that? This is a bit of a lazy question as I'm not familiar with `DispatchSourceTimer`.

Reply:
I am imagining that we could have a test for this where the first flush interval is long, like 10 seconds, and then we set a new very short flush interval, like 0.05 seconds. Is the expectation that the handler would only be invoked even if more than 10 seconds have passed?